src/ConstructiveSolidGeometry/SurfacePrimitives/Rectangle.jl

```diff
@@ -0,0 +1,104 @@
+struct Rectangle{T,TN,TL,TW} <: AbstractSurfacePrimitive{T}
```

Comment:
How about changing the order to `Rectangle{TN, T, TL, TW}`.
This would allow to easily introduce aliases like `RectangleX{T,TL,TW} = Rectangle{Val(:x),T,TL,TW}` which might be more convenient for the user (see `RegularPrism`).

Reply:
Also, we do not need to save `TN` as a field to the struct, but just leave it as a parametric type, right?